Gems/Atom/RHI/Vulkan/Code/Source/RHI/ShaderResourceGroupPool.cpp

```diff
@@ -34,15 +34,21 @@ namespace AZ
 
             m_descriptorSetCount = RHI::Limits::Device::FrameCountMax;
 
+#if !defined(AZ_PLATFORM_ANDROID)
```

Comment:
Why not do this for all vulkan platforms?

Reply:
This use of the AZ_PLATFORM_* macro is also causing the change to fail vaildation, I believe we want platform specific code to be moved over to the platform abstraction layer. 
Platform specific code for the vulkan RHI is placed here - Gems/Atom/RHI/Vulkan/Code/Include/Platform & Gems/Atom/RHI/Vulkan/Code/Source/Platform.

Reply:
done.
`AZ_PLATFORM_ANDROID` has been removed.